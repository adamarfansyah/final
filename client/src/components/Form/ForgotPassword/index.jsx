import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import { FormattedMessage } from 'react-intl';

import forgotPasswordImage from '@static/images/forgot-password.svg';
import Button from '@components/Button';
import Input from '@components/Input';
import classes from './style.module.scss';

const ForgotPasswordForm = ({ onSubmit }) => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  return (
    <div className={classes.formEmail}>
      <img className={classes.image} src={forgotPasswordImage} alt="forgot-password" />
      <div className={classes.form}>
        <div className={classes.title}>
          <FormattedMessage id="forgot_password" />
        </div>
        <form onSubmit={handleSubmit(onSubmit)}>
          <div>
            <Input
              type="text"
              name="email"
              label="Email"
              errors={errors}
              register={register}
              placeholder="E-mail"
              validationSchema={{
                required: 'Email is required',
                pattern: { value: /^[^\s@]+@[^\s@]+\.[^\s@]+$/, message: 'Invalid email address' },
              }}
              required
            />
          </div>
          <Button type="submit" className={classes.btn}>
            <FormattedMessage id="app_submit" />
          </Button>
        </form>
      </div>
    </div>
  );
};

ForgotPasswordForm.propTypes = {
  onSubmit: PropTypes.func,
};

export default ForgotPasswordForm;
