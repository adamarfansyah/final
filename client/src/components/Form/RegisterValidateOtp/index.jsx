import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { FormattedMessage } from 'react-intl';
import { useNavigate } from 'react-router-dom';

import Countdown from 'react-countdown';
import Button from '@components/Button';
import Input from '@components/Input';
import classes from './style.module.scss';

const RegisterValidateOtp = ({ user, onSubmit, time, resendOtp, blockUrl }) => {
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  useEffect(() => {
    if (!user) {
      navigate(blockUrl);
    }
  }, [user]);

  const handleResend = () => {
    resendOtp(user.email);
  };

  return (
    <div className={classes.form}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div>
          <Input
            type="text"
            name="otp"
            label="OTP"
            errors={errors}
            register={register}
            validationSchema={{
              required: 'OTP is required',
              minLength: { value: 3, message: 'OTP must be at least 3 characters' },
            }}
            placeholder="OTP"
            required
          />
          <Button type="submit">
            <FormattedMessage id="app_submit" />
          </Button>
        </div>
      </form>
      <div className={classes.times}>
        <Countdown date={time}>
          <Button variant="secondary" className={classes.btnRetry} onClick={handleResend}>
            <FormattedMessage id="resend_otp" />
          </Button>
        </Countdown>
      </div>
    </div>
  );
};

RegisterValidateOtp.propTypes = {
  user: PropTypes.object,
  onSubmit: PropTypes.func,
  time: PropTypes.string,
  resendOtp: PropTypes.func,
  blockUrl: PropTypes.string,
};

export default RegisterValidateOtp;
