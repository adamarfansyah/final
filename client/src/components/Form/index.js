import Login from './Login';
import Register from './Register';
import RegisterValidateOtp from './RegisterValidateOtp';
import ForgotPasswordForm from './ForgotPassword';

export { Login, Register, RegisterValidateOtp, ForgotPasswordForm };
