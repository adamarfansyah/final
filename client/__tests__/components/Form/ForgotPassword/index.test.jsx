import { render, act, fireEvent } from '@utils/testHelper';
import { ForgotPasswordForm } from '@components/Form';

const mockOnSubmit = jest.fn((email) => Promise.resolve({ email }));

let wrapper;

beforeEach(() => {
  wrapper = render(<ForgotPasswordForm onSubmit={mockOnSubmit} />);
});

describe('Component Forgot Password', () => {
  test('should render forgot password button and form correct', () => {
    const { getByText, getByTestId } = wrapper;
    const button = getByTestId('btn');
    const labelEmail = getByText(/email/i);

    expect(labelEmail).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });

  test('Should Input email have value', () => {
    const { getByPlaceholderText, getByTestId } = wrapper;
    const input = getByPlaceholderText('E-mail');
    const button = getByTestId('btn');

    act(() => {
      fireEvent.change(input, {
        target: { value: 'test@example.com' },
      });
    });

    expect(button).toBeInTheDocument();

    act(() => {
      fireEvent.submit(button);
    });

    expect(input).toHaveValue('test@example.com');
  });

  test('Should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
