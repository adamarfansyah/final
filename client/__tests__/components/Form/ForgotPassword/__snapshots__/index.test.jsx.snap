// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Forgot Password Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="formEmail"
      >
        <img
          alt="forgot-password"
          class="image"
        />
        <div
          class="form"
        >
          <div
            class="title"
          >
            Lupa Password
          </div>
          <form>
            <div>
              <div
                class="input"
                data-testid="input-container"
              >
                <div
                  class="label"
                  data-testid="input-label"
                >
                  Email
                  <span>
                    *
                  </span>
                </div>
                <div
                  class="contentInput"
                >
                  <input
                    data-testid="input"
                    id="email"
                    name="email"
                    placeholder="E-mail"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <button
              class="btn btn"
              data-testid="btn"
              type="submit"
            >
              Kirim
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="formEmail"
    >
      <img
        alt="forgot-password"
        class="image"
      />
      <div
        class="form"
      >
        <div
          class="title"
        >
          Lupa Password
        </div>
        <form>
          <div>
            <div
              class="input"
              data-testid="input-container"
            >
              <div
                class="label"
                data-testid="input-label"
              >
                Email
                <span>
                  *
                </span>
              </div>
              <div
                class="contentInput"
              >
                <input
                  data-testid="input"
                  id="email"
                  name="email"
                  placeholder="E-mail"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <button
            class="btn btn"
            data-testid="btn"
            type="submit"
          >
            Kirim
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
